src/wazuh_modules/wm_syscheck.c

```diff
@@ -0,0 +1,271 @@
+#endif
+        }
+
+        r++;
+    }
+}
+
+static void wm_syscheck_log_config(const wm_syscheck_t *sys)
+{
+    cJSON * config_json = wm_syscheck_dump(sys);
+    if (config_json) {
+        char * config_str = cJSON_PrintUnformatted(config_json);
+        if (config_str) {
+            mtdebug1(WM_SYSCHECK_LOGTAG, "%s", config_str);
+            cJSON_free(config_str);
+        }
+        cJSON_Delete(config_json);
+    }
+}
+
+void* wm_syscheck_main(wm_syscheck_t *sys) {
+    syscheck_config* config = (syscheck_config*)sys;
+    syscheck = *config;
+    read_internal(0);
+    mtdebug1(WM_SYSCHECK_LOGTAG, "Starting syscheck.");
+    if (syscheck.disabled == 1) {
+        if (!syscheck.dir) {
+            mtinfo(WM_SYSCHECK_LOGTAG, FIM_DIRECTORY_NOPROVIDED);
+            dump_syscheck_file(&syscheck, "", 0, NULL, 0, NULL, NULL, -1);
+        } else if (!syscheck.dir[0]) {
+                mtinfo(WM_SYSCHECK_LOGTAG, FIM_DIRECTORY_NOPROVIDED);
+        }
+        os_free(syscheck.dir[0]);
+        if (!syscheck.ignore) {
+            os_calloc(1, sizeof(char *), syscheck.ignore);
+        } else {
+            os_free(syscheck.ignore[0]);
+        }
+#ifdef WIN32
+        if (!syscheck.registry) {
+            dump_syscheck_registry(&syscheck, "", 0, NULL, NULL,  0, NULL, 0, -1);
+        }
+        os_free(syscheck.registry[0].entry);
+#endif //WIN32
+
+        mtinfo(WM_SYSCHECK_LOGTAG, FIM_DISABLED);
+        mtinfo(WM_SYSCHECK_LOGTAG, "Module disabled. Exiting...");
+        pthread_exit(NULL);
+    }
+    wm_syscheck_log_config(sys);
+    /* Rootcheck config */
+    syscheck.rootcheck = !rootcheck_init(0);
+#ifndef WIN32
+    // Start com request thread
+    w_create_thread(syscom_main, NULL);
+
+    if (syscheck.rootcheck) {
+        rootcheck_connect();
+    }
+    /* Connect to the queue */
+    if ((syscheck.queue = StartMQ(DEFAULTQUEUE, WRITE, INFINITE_OPENQ_ATTEMPTS)) < 0) {
+        mterror_exit(WM_SYSCHECK_LOGTAG, QUEUE_FATAL, DEFAULTQUEUE);
+    }
+#endif //WIN32
+    wm_syscheck_print_info();
+    fim_initialize();
+#ifndef WIN32
+    // Audit events thread
+    if (syscheck.enable_whodata) {
+#ifdef ENABLE_AUDIT
+        int out = audit_init();
+        if (out < 0) {
+            mtwarn(WM_SYSCHECK_LOGTAG, FIM_WARN_AUDIT_THREAD_NOSTARTED);
+
+            // Switch who-data to real-time mode
+
+            for (int i = 0; syscheck.dir[i] != NULL; i++) {
+                if (syscheck.opts[i] & WHODATA_ACTIVE) {
+                    syscheck.opts[i] &= ~WHODATA_ACTIVE;
+                    syscheck.opts[i] |= REALTIME_ACTIVE;
+                }
+            }
+        }
+#else
+        mterror(WM_SYSCHECK_LOGTAG, FIM_ERROR_WHODATA_AUDIT_SUPPORT);
+#endif
+    }
+#endif //WIN32
+    /* Start the daemon */
+    start_daemon();
+    mtinfo(WM_SYSCHECK_LOGTAG, "Module finished.");
+    return 0;
+}
+
+void wm_syscheck_destroy(wm_syscheck_t *data) {
+    mtinfo(WM_SYSCHECK_LOGTAG, "Destroy received for Syscheck.");
```

Comment:
this function is called in some moment?

Reply:
should be called as part of wmoduels graceful shutdown.